### PR TITLE
DDF-04522 migrate metacard-quality to react

### PR DESCRIPTION
**Pull request by andrewzimmer**
#### What does this PR do?
Migrate metacard-quality to React.
#### Who is reviewing it? 

@Bdthomson @hayleynorton @nsuvarna @lamhuy 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler 
@djblue 

#### How should this be tested?
- build/run
- verify no regressions in the metacard-quality component
#### What are the relevant tickets?

For GH Issues:
Fixes: #4522  

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by Schachte** [comment]
https://github.blog/2019-02-14-introducing-draft-pull-requests/

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-quality/index.tsx [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default } from './metacard-quality'
```
Comment (hayleynorton) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-quality/index.tsx]: ✏️ codice header

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-quality/metacard-quality.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      attributeValidation: [],
+      metacardValidation: [],
+      loading: true,
+    }
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
+      const attributeValidationRes = fetch(
+        `./internal/metacard/${this.state.model
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-quality/metacard-quality.tsx]: ✏️ 	Pull this out into a constant.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/metacard-quality/metacard-quality.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  }
+
+  tbody {
+    border-top: 1px solid rgba(100, 100, 100, 0.3);
+  }
+`
+
+const render = (props: Props) => {
+  const { metacardValidation, attributeValidation, loading } = props
+  return (
```
Comment (Bdthomson) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/metacard-quality/metacard-quality.tsx]: This is a very large render. I'd recommend pulling out what you can into named functions that return jsx.

**Comment by andrewzimmer** [comment]
build now

**Comment by Bdthomson** [comment]
Hero in progress

**Comment by Bdthomson** [comment]
Before
image

After
image

Hero Successful, but I would like to request that you make a small style change while you are in here.

Requested Change
```javascript
const Header = styled.h4`
  text-align: left;
  padding: ${props => props.theme.minimumSpacing}
`
```

Adding padding here makes it look a lot nicer.

**Comment by andrewzimmer** [comment]
build now

**Comment by andrewzimmer** [comment]
build now

**Pull request merged**